cartridges/int_algolia_sfra/cartridge/controllers/Cart.js

```diff
@@ -24,4 +26,45 @@ server.append('Show', function (req, res, next) {
     next();
 });
 
+
+server.append('AddProduct', function (req, res, next) {
+    if (algoliaData.getPreference('Enable') && algoliaData.getPreference('EnableInsights')) {
+
+        var productId = req.form.pid;
+
+        var algoliaProductData = {};
+        algoliaProductData.pid = productId;
+
+        if (algoliaData.getPreference('RecordModel') === MASTER_LEVEL) {
+            var product = ProductMgr.getProduct(productId);
+
+            if (product && (product.isMaster() || product.isVariationGroup())) {
+                algoliaProductData.pid = product.masterProduct.ID;
+            }
```

Comment:
This is still not good, the `product` is NOT a master product, it's a variation product, so it never passes in that condition. Please test it, you can inspect with the Browser's Network console and see that it's the variantID that is sent instead of the masterID
```suggestion
            algoliaProductData.pid = product.masterProduct.ID;
```